### ISSUE TITLE
tcp_drop is not traceable even with kernel verion 5.15.0

### ISSUE THREAD

**Issue by paulzhn** [iovisor/bpftrace#2448]
### What reproduces the bug?

Running the tcpdrop.bt, and the result is 

```
./tcpdrop.bt:41-43: WARNING: tcp_drop is not traceable (either non-existing, inlined, or marked as "notrace"); attaching to it will likely fail
Attaching 3 probes...
ERROR: Could not resolve symbol: /proc/self/exe:BEGIN_trigger
```

I found the pr #2379 was merged and it fixed the tcp_drop problem for version > 5.17. However, it seems like not only version > 5.17 kernel has no tcp_drop. My OS is Ubuntu 22.04.1 LTS and kernel version is 5.15.0-52-generic, but still doesn't have this probe. And the command `cat /proc/kallsyms | grep tcp_drop` returns nothing. 

I found an issue from redhat https://bugzilla.redhat.com/show_bug.cgi?id=1907357, and it said with the kernel version 5.9, the probe could not be found either. I am not familiar with compiler, but I think maybe it's related to it? 

### bpftrace --info

```
System
  OS: Linux 5.15.0-52-generic #58-Ubuntu SMP Thu Oct 13 08:03:55 UTC 2022
  Arch: x86_64

Build
  version: v0.14.0
  LLVM: 11.1.0
  ORC: v2
  foreach_sym: yes
  unsafe uprobe: no
  bfd: no
  bpf_attach_kfunc: yes
  bcc_usdt_addsem: yes
  bcc bpf_attach_uprobe refcount: yes
  bcc library path resolution: yes
  libbpf: yes
  libbpf btf dump: yes
  libbpf btf dump type decl: yes
  libdw (DWARF support): no

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: yes
  get_boot_ns: yes
  dpath: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf (depends on Build:libbpf): yes
  map batch (depends on Build:libbpf): yes
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
  iter:task: yes
  iter:task_file: yes
```

**Comment by viktormalik** [comment]
Hi,

> I found the pr #2379 was merged and it fixed the tcp_drop problem for version > 5.17. However, it seems like not only version > 5.17 kernel has no tcp_drop. My OS is Ubuntu 22.04.1 LTS and kernel version is 5.15.0-52-generic, but still doesn't have this probe. And the command `cat /proc/kallsyms | grep tcp_drop` returns nothing.
> 
> I found an issue from redhat https://bugzilla.redhat.com/show_bug.cgi?id=1907357, and it said with the kernel version 5.9, the probe could not be found either. I am not familiar with compiler, but I think maybe it's related to it?

`tcp_drop` has been removed in 5.19, but GCC has been inlining it for some time already, so that's probably your issue. 

You could try to use the new version of [tcpdrop.bt](https://github.com/iovisor/bpftrace/blob/master/tools/tcpdrop.bt) although it may not work as it uses some fields of `tracepoint:skb:kfree_skb` that were only introduced in 5.17 (in particular the `reason` field). Perhaps you could try to tweak it so that it doesn't need that field, but other than that, I don't see any other options.

**Comment by viktormalik** [comment]
@paulzhn can we close this or you need more assistance?
